### PR TITLE
Runs migration script

### PR DESCRIPTION
**Pull request by SiddhantSadangi**
# Description

Initial version of script used to migrate runs from one project to another

__Related to:__ Project run migrator

__Any expected test failures?__

---

Add a `[X]` to relevant checklist items

## ❔ This change

- [X] adds a new feature
- [ ] fixes breaking code
- [ ] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

- [X] Refactored code ([sourcery](https://sourcery.ai/))
- [X] Tested code locally
- [X] Precommit installed and run before pushing changes
- [X] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [X] Updated GitHub [README](../README.md)
- [ ] Updated the projects overview page on Notion

---

## 🧪 Test Configuration

- OS: Windows11
- Python version: 3.8.15
- Neptune version: 1.3.1
- Affected libraries with version:

### CODE REVIEW

File: utils/migration_tools/from_another_project/runs_migrator.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+# %%
+from_project = (
+    input("Enter project name to migrate from in WORKSPACE_NAME/PROJECT_NAME format:")
+    .strip()
+    .lower()
+)
+
+# %%
+to_project = (
```
Comment (Blaizzy) [inline utils/migration_tools/from_another_project/runs_migrator.py:49]: How about a flag to delete "from_project" runs/metadata once it's done copying?

File: utils/migration_tools/from_another_project/runs_migrator.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+            capture_traceback=False,
+            git_ref=GitRef.DISABLED,
+            source_files=[],
+        ) as to_run:
+            to_run_id = to_run["sys/id"].fetch()
+            logging.info(f"Copying {from_project}/{from_run_id} to {to_project}/{to_run_id}")
+
+            namespaces = flatten_namespaces(from_run.get_structure())
+
+            for namespace in namespaces:
```
Comment (Blaizzy) [inline utils/migration_tools/from_another_project/runs_migrator.py:164]: Probably add warnings for when copying from a run's namespace fails.

**Review by Blaizzy — CHANGES_REQUESTED** [review]
Looks good!

I just have minor suggestions.

File: utils/migration_tools/from_another_project/runs_migrator.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    .lower()
+)
+
+# %%
+to_project = (
+    input("Enter project name to migrate to in WORKSPACE_NAME/PROJECT_NAME format:").strip().lower()
+)
+
+assert to_project != from_project, "To and from projects need to be different"
+# %%
```
[the remark this answers is not in the document]
Comment (SiddhantSadangi) [reply utils/migration_tools/from_another_project/runs_migrator.py:54]: That's mentioned as a prerequisite in the README: https://github.com/neptune-ai/examples/pull/251/files#diff-6033e237f6e1281e2e01936ee288e1e6c175e99d19b67ef422155be2dd4cbefeR7

File: utils/migration_tools/from_another_project/runs_migrator.py (hunk 1 above)
Comment (SiddhantSadangi) [reply utils/migration_tools/from_another_project/runs_migrator.py:49]: That would be a good addon. But given that it is already pretty easy to delete all the runs in a project, I would defer adding it to this script to a later version based on user feedback

File: utils/migration_tools/from_another_project/runs_migrator.py (hunk 2 above)
Comment (SiddhantSadangi) [reply utils/migration_tools/from_another_project/runs_migrator.py:164]: Already handled here: https://github.com/neptune-ai/examples/pull/251/files#diff-d07e1d383cb63514632d1083ccc4b1fcb131d9ca42125ead53a6d4bcbd3da6e8R215-R217

File: utils/migration_tools/from_another_project/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+## Note
+
+There are a few things to keep in mind when using this script:
+
+- Currently, only run metadata is copied. Project and model metadata are not copied.
+- Most of the namespaces from the original runs will be retained in the new runs, except for the following:
+  - `sys` namespaces:
+    - The `state` namespace cannot be copied.
+    - The `description`, `name`, `custom_run_id`, and `tags` namespaces are copied to the `sys` namespace in the new run.
+    - All other namespaces are copied to a new `old_sys` namespace in the new run.
```
Comment (normandy7) [inline utils/migration_tools/from_another_project/README.md:28]:
```suggestion
    - The `state` field cannot be copied.
    - The `description`, `name`, `custom_run_id`, and `tags` fields are copied to the `sys` namespace in the new run.
    - All other fields are copied to a new `old_sys` namespace in the new run.
```

**Review by Blaizzy — APPROVED** [review]
LGTM

**Review by normandy7 — APPROVED** [review]

**Pull request merged**